subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ivy/IvyDynamicRevisionRemoteResolveIntegrationTest.groovy

```diff
@@ -1228,6 +1228,7 @@ configurations.all {
         expectGetDynamicRevision(httpModule)
 
         then:
+        args '--refresh-dependencies'
```

Comment:
I wonder if this test would be clearer if we used a different module for each of the 3 cases, rather than incrementally adding and then needing `--refresh-dependencies` for the last one.
The "missing from local", "missing from remote" and "present in both" conditions are not really a sequence of changes to a repository, but are instead 3 independent use cases.